backend/Tournament/views.py

```diff
@@ -0,0 +1,131 @@
+		
+class TournamentDetailView(APIView):
+	def get(self, request, tournament_id):
+		tournament = Tournament.objects.filter(id=tournament_id).first()
+		if not tournament:
+			return Response(f"Tournament {tournament_id} not found.", status=status.HTTP_404_NOT_FOUND)
+		serializer = TournamentSerializer(tournament)
+		return Response(serializer.data, status=status.HTTP_200_OK)
+
+	@method_decorator(must_be_authenticated)
+	@method_decorator(check_that_valid_tournament_request([TournamentState.LOBBY, TournamentState.IN_PROGRESS]))
+	def post(self, request, tournament_id):
+		'''If tournament is in lobby, start tournament.
+
+		If tournament is in progress, return next match url.
+		'''
+		tournament = Tournament.objects.get(id=tournament_id)
+		if tournament.state == TournamentState.LOBBY:
+			try:
+				TournamentSetupManager.start_tournament(tournament)
+				serializer = TournamentInProgressSerializer(tournament)
+				return Response(serializer.data, status=status.HTTP_200_OK)
+			except Exception as e:
+				return Response({'error': str(e)}, status=status.HTTP_403_FORBIDDEN)
+
+		elif tournament.state == TournamentState.IN_PROGRESS:
+			pass
+
+	@method_decorator(must_be_authenticated)
+	@method_decorator(check_that_valid_tournament_request([TournamentState.LOBBY, TournamentState.IN_PROGRESS]))
+	def delete(self, request, tournament_id):
+		tournament = Tournament.objects.filter(id=tournament_id).first()
+		if not tournament:
+			return Response(f"Tournament {tournament_id} not found.", status=status.HTTP_404_NOT_FOUND)
+		tournament.delete()
+		return Response(status=status.HTTP_204_NO_CONTENT)
+
+class TournamentPlayerListView(APIView):
+	@method_decorator(must_be_authenticated)
+	@method_decorator(check_that_valid_tournament_request([TournamentState.LOBBY]))
+	def post(self, request, tournament_id):
+		username = request.data.get('username', None)
+		password = request.data.get('password', None)
+		display_name = request.data.get('display_name', None)
+		if not username or not password:
+			return Response({"error": "Username and password are required"}, status=status.HTTP_400_BAD_REQUEST)
+		user = authenticate(username=username, password=password)
+		if user is None:
+			return Response({"message": "Invalid username or password"}, status=status.HTTP_401_UNAUTHORIZED)
+
+		tournament = Tournament.objects.get(id=tournament_id)
+		if tournament.players.count() >= tournament.player_amount:
+			return Response({"error": "Tournament already has maximum number of players"}, status=status.HTTP_400_BAD_REQUEST)
+
+		try:
+			tournament_player = TournamentPlayer.objects.create(
+				tournament=tournament,
+				user=user,
+				display_name=display_name,
+			)
+			tournament_player_serializer = TournamentPlayerSerializer(tournament_player)
+			return Response(tournament_player_serializer.data, status=status.HTTP_201_CREATED)
+		except Exception as e:
+			return Response({'error': str(e)}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
And please check all of the other errors too, forbidden is very rearly the correct status code.